src/python_testing/TC_SU_2_2.py

```diff
@@ -0,0 +1,844 @@
+        ./out/debug/chip-ota-requestor-app \
+            --discriminator {discriminator} \
+            --passcode {passcode} \
+            --secured-device-port {secured_device_port} \
+            --autoApplyImage \
+            --KVS /tmp/chip_kvs_requestor; exit"
+        end tell'
+        '''
+        proc = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        await asyncio.sleep(10)
+        logger.info(f'Launched OTA Requestor with command: {cmd}')
+        return proc
+
+    async def set_ota_acls_for_provider(self, controller, requestor_node: int, provider_node: int, fabric_index: int):
+        """
+        Set ACLs for OTA interaction between a Requestor and Provider.
+        Preserves existing ACLs to avoid overwriting.
+        """
+        # -------------------------------
+        # Prerequisite #3.1: ACLs on Requestor to allow Provider to interact with OTA Requestor
+        # -------------------------------
+
+        # Read existing ACLs on Requestor
+        acl_original_requestor = await self.read_single_attribute(
+            dev_ctrl=controller,
+            node_id=requestor_node,
+            endpoint=0,
+            attribute=Clusters.AccessControl.Attributes.Acl,
+        )
+        logger.info(f'Prerequisite #3.1 - ACLs current privileges on Requestor (DUT): {acl_original_requestor}')
+
+        # Add minimal ACL: ACLs on Requestor to allow Provider interaction
+        acl_operate_provider = Clusters.AccessControl.Structs.AccessControlEntryStruct(
+            fabricIndex=fabric_index,
+            privilege=Clusters.AccessControl.Enums.AccessControlEntryPrivilegeEnum.kOperate,
+            authMode=Clusters.AccessControl.Enums.AccessControlEntryAuthModeEnum.kCase,
+            subjects=[provider_node],
+            targets=[Clusters.AccessControl.Structs.AccessControlTargetStruct(
+                endpoint=0,
+                cluster=Clusters.OtaSoftwareUpdateRequestor.id)]
+        )
+
+        # Combine original + new ACLs on Requestor
+        # This avoids overwriting existing entries when adding new ACLs
+        # acls_requestor = acl_original_requestor + [acl_operate_provider]
+        acls_requestor = acl_original_requestor + [acl_operate_provider]
+        resp = await self.write_acl(controller, requestor_node, acls_requestor)
+        logger.info(
+            f'Prerequisite #3.1 - Response of wrote combined ACLs on Requestor to allow Provider with node {provider_node}: {resp}')
+
+        # Read new ACLs on Requestor (after combined)
+        acl_current_requestor = await self.read_single_attribute(
+            dev_ctrl=controller,
+            node_id=requestor_node,  # DUT
+            endpoint=0,
+            attribute=Clusters.AccessControl.Attributes.Acl,
+        )
+        logger.info(f'Prerequisite #3.1 - ACLs combined current + new privileges on Requestor (DUT): {acl_current_requestor}')
+
+        # -------------------------------
```

Comment:
You should end up getting this for free once the Alex PR lands, I think (https://github.com/project-chip/connectedhomeip/pulls/asirko-soft)